source/core/parameters.cc

```diff
@@ -813,6 +813,114 @@ namespace Parameters
     prm.leave_subsection();
   }
 
+  template <int dim>
+  void
+  Laser<dim>::declare_parameters(ParameterHandler &prm)
+  {
+    prm.enter_subsection("laser parameters");
+    {
+      prm.declare_entry("activate laser",
+                        "false",
+                        Patterns::Bool(),
+                        "Activate laser");
+      prm.declare_entry("concentration factor",
+                        "2.0",
+                        Patterns::Double(),
+                        "Concentration factor");
+      prm.declare_entry("laser power",
+                        "0.0",
+                        Patterns::Double(),
+                        "Laser power");
+      prm.declare_entry("laser absorptivity",
+                        "0.6",
+                        Patterns::Double(),
+                        "Laser absorptivity");
+      prm.declare_entry("penentration depth",
+                        "0.0",
+                        Patterns::Double(),
+                        "Penentration depth");
+      prm.declare_entry("beam radius",
+                        "0.0",
+                        Patterns::Double(),
+                        "Laser beam radius");
+
+
+      prm.enter_subsection("path");
+      laser_path = std::make_shared<Functions::ParsedFunction<dim>>(dim);
+      laser_path->declare_parameters(prm, dim);
+      if (dim == 2)
+        prm.set("Function expression", "0; 0");
+      if (dim == 3)
+        prm.set("Function expression", "0; 0; 0");
+      prm.leave_subsection();
+
+      prm.declare_entry("start time",
+                        "0.0",
+                        Patterns::Double(),
+                        "Start time of laser");
+      prm.declare_entry("end time",
+                        "0.0",
+                        Patterns::Double(),
+                        "End time of laser");
+
+      prm.declare_entry("beam direction",
```

Comment:
I think "orientation" would be a better choice of word than direction here.

Reply:
changed to orientation